analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -196,6 +196,8 @@ name switch
                     // that can not bind to a type. The only things that can bind to a type are SimpleNames (Identifier or GenericName)
                     // or pre-defined types. None of the pre-defined types have a nested type, so we can exclude these as well.
                     { Parent: MemberAccessExpressionSyntax x } when AnyMemberAccessLeftIsNotASimpleName(x) => TokenType.UnknownTokentype,
+                    // The left side of a pointer member access must be a pointer and can not be a type
```

Comment:
Educational: this assertion still holds when using `using unsafe` from C# 12, right?

```cs
using System;
using unsafe Ptr = int*;

internal class Program
{
    private unsafe static void Main(string[] args)
    {
        int x = 3;
        Ptr y = &x;
        Console.WriteLine($"x = {y->ToString()}");
    }
}
```
`Ptr` should still be considered `TokenType.UnknownTokentype`, and not a type. Correct?

Reply:
It just works. `Ptr` is in a type position and is therefore classified as type (There is a distinguishing happening before `ClassifyMemberAccess` is called. An IdentfierSymtax` can either be in an expression or a type context. In a type context, a single IdentfierName resolves to TokenType.TypeName and in an expression context to UnknownTokentype).